### ISSUE TITLE
CI failing with `--no-default-features`

### ISSUE THREAD

**Issue by curquiza** [meilisearch/meilisearch#4441]
**Describe the bug**
CI runing with `cargo build --locked --release --no-default-features --all` fails on `release-v1.7.0`

**Expected behavior**
No CI failing

**Screenshots**
https://github.com/meilisearch/meilisearch/actions/runs/8045912784/job/21972108739

<img width="934" alt="Capture d’écran 2024-02-26 à 09 46 43" src="https://github.com/meilisearch/meilisearch/assets/20380692/b1de101b-ff61-4a79-b981-ad78df319be2">

**Meilisearch version:**
Failing on `release-v1.7.0`.

**Comment by dureuill** [comment]
Hello @curquiza :wave:

I do not reproduce on branch release-v1.7.0 (`147a67dc8266c77479d86f13a2d7ac0d7054c135`), on an Ubuntu 18.04 (docker).

Command I ran:

```
RUSTFLAGS="-D warnings" cargo build --release --locked --no-default-features --al
```

**Comment by curquiza** [comment]
The problem is the CI is not working on `release-v1.7.0` as I linked in the issue 😢 
https://github.com/meilisearch/meilisearch/actions/runs/8045912784

Here is the CI manifest: https://github.com/meilisearch/meilisearch/blob/main/.github/workflows/test-suite.yml

**Comment by dureuill** [comment]
Fixed by #4451
